### PR TITLE
Update opam files

### PR DESCRIPTION
**Pull request by lindig**
The opam files have been updated using "dune external-lib-deps" to find the real dependencies.

Signed-off-by: Christian Lindig <christian.lindig@citrix.com>

### CODE REVIEW

File: xapi-database.opam [hunk 1]
```diff
@@ -9,14 +9,24 @@ build: [[ "jbuilder" "build" "-p" name ]]
...
+  "ppx_deriving_rpc"
   "ppx_sexp_conv"
-  "xapi-libs-transitional"
-  "xapi-idl"
+  "rpclib"
+  "sexpr"
+  "base-threads"
+  "uuid"
+  "xapi-consts"
+  "xapi-datamodel"
```
Comment (edwintorok) [inline xapi-database.opam]: This causes a loop and Travis complains:
```
The actions to process have cyclic dependencies:
  - install xapi-database.master -> install xapi-datamodel.master -> install xapi-database.master
  - install xapi-datamodel.master -> install xapi-consts.master -> install xapi-datamodel.master
  - install xapi-database.master -> install xapi-datamodel.master -> install xapi-consts.master -> install xapi-database.master
```

File: xapi.opam [hunk 2]
```diff
@@ -10,38 +10,52 @@ build-test: [[ "jbuilder" "runtest" "-p" name "-j" jobs ]]
...
+  "xapi-stdext-unix"
   "xapi-tapctl"
   "xapi-test-utils"
   "xapi-types"
   "xapi-xenopsd"
+  "xcp"
+  "xcp-inventory"
   "xenctrl"
-  "xenstore"
+  "xml-light2"
```
Comment (edwintorok) [inline xapi.opam]: This seems to come from perftest, but this is what the dune file says, so updating it in opam is correct, we can look at pruning the dependencies from dune later if needed.

File: xapi-database.opam [hunk 3]
```diff
@@ -9,14 +9,22 @@ build: [[ "jbuilder" "build" "-p" name ]]
 
 depends: [
   "jbuilder" {build & >= "1.0+beta11"}
-  "rpc"
+
+  "gzip"
+  "http-svr"
+  "ocaml-migrate-parsetree"
```
Comment (mseri) [inline xapi-database.opam]: Omp appears in multiple packages but it should not, maybe something is wrong in some dune/jbuilder file

Comment (edwintorok) [reply xapi-database.opam]: We should keep an eye out when updating the other packages, eventually we should find the one responsible for it.

**Pull request merged**